src/Client/Updater.php

```diff
@@ -167,12 +171,14 @@ public function refresh(bool $force = false): bool
     /**
      * Updates the timestamp role, per section 5.3 of the TUF spec.
      */
-    private function updateTimestamp(): TimestampMetadata
+    private function updateTimestamp(): ?TimestampMetadata
     {
         // § 5.4.1
-        $newTimestampData = $this->server->getTimestamp();
-
-        $this->universalVerifier->verify(TimestampMetadata::TYPE, $newTimestampData);
+        /** @var \Tuf\Metadata\TimestampMetadata $newTimestampData */
+        $newTimestampData = $this->universalVerifier->verify(TimestampMetadata::TYPE, $this->server->getTimestamp());
+        if ($newTimestampData === $this->storage->getTimestamp()) {
```

Comment:
Looking at https://github.com/theupdateframework/specification/pull/209 which is one of the commits for the updated tag

> timestamp should be updated only when there's a newer version AND the referenced snapshot version is not less than the one in the already trusted timestamp

Are we comparing the versions of the snapshot referenced in the new and existing timestamps?